cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/LocalApplicationDetailFetcher.java

```diff
@@ -56,7 +56,7 @@ public LocalApplicationDetailFetcher(ApplicationLifecycleService applicationLife
   public ApplicationDetail get(ApplicationId appId) throws IOException, NotFoundException {
     ApplicationDetail detail = null;
     try {
-      detail = applicationLifecycleService.getAppDetail(appId);
+      detail = applicationLifecycleService.getLatestAppDetail(appId.getAppReference());
```

Comment:
This is changing the existing behavior, why is it needed? If it is needed, the ApplicationDetailFetcher should be changed to take an ApplicationReference instead of an id.

Reply:
I see it's only called by some support bundle thing, I think the right thing to do is change the interface to pass in a reference.